### PR TITLE
fix: do not create resources for marked deleted tenant

### PR DESCRIPTION
**Pull request by jiuker**
fix: will not create anything for marked deleted tenant

### CODE REVIEW

**Review by shtripat — DISMISSED** [review]
LGTM. Can we have steps to verify the changes locally?

**Comment by pjuarezd** [comment]
> > LGTM. Can we have steps to verify the changes locally?
> 
> You can deploy tenant with finalizer. And then delete it @shtripat

which finalizer @jiuker?

File: pkg/controller/main-controller.go [hunk 1]
```diff
@@ -804,6 +804,12 @@ func (c *Controller) syncHandler(key string) (Result, error) {
 		// will retry after 5sec
 		return WrapResult(Result{RequeueAfter: time.Second * 5}, nil)
 	}
+	// Check if the Tenant is marked to be deleted
+	// Shouldn't create resources when marked for deletion
+	if !tenant.DeletionTimestamp.IsZero() {
+		runtime.HandleError(fmt.Errorf("Tenant '%s' is marked for deletion", key))
```
Comment (pjuarezd) [inline pkg/controller/main-controller.go:810]:
```suggestion
		runtime.HandleError(fmt.Errorf("Tenant '%s' is marked for deletion, skipping", key))
```

**Review by pjuarezd — CHANGES_REQUESTED** [review]

**Review by pjuarezd — APPROVED** [review]

**Review by shtripat — APPROVED** [review]
lgtm

**Pull request merged**